Plugins/NativeEngine/Source/NativeEngine.cpp

```diff
@@ -266,7 +267,7 @@ namespace Babylon
                     image = bimg::imageConvert(&allocator, bimg::TextureFormat::RGBA8, *image, false);
                     bimg::imageFree(oldImage);
                 }
-                else if (image->m_format == bimg::TextureFormat::RGBA16)
+                else if (image->m_format == bimg::TextureFormat::RGBA16 || image->m_format == bimg::TextureFormat::R16)
```

Comment:
I guess this quadruples the size of the texture with a lot of wasted data? Is it possible to support R16 more directly?

Reply:
more than quadruple: *8 !! It's possible (I guess) to update bimg downsample to support R16.

Reply:
bgfx only supports generating mips of certain formats. Maybe that can be updated to support more formats.

Reply:
Wait maybe I'm confused... isn't R16 16 bits per pixel, and RGB32F is 32 bits per pixel, so we are actually doubling the storage but losing half the precision (8 bit red channel instead of 16 bit red channel)? Someone help me with my 5th grade math plz. 🤣

Reply:
Definitely! I'm waiting for feedback from the user and I'll continue the work accordingly.

Reply:
> Wait maybe I'm confused... isn't R16 16 bits per pixel, and RGB32F is 32 bits per pixel, so we are actually doubling the storage but losing half the precision (8 bit red channel instead of 16 bit red channel)? Someone help me with my ~5th~ ~4th~ 3rd grade math plz. 🤣

R16 is 16bits per pixel or 2 bytes per texel. RGBA32F is 4 floats or 4 * 4bytes = 16 bytes. RGBA8 is indeed 32bits